### ISSUE TITLE
Unable to resume activity {com.abjjad.app/com.facebook.FacebookActivity}:

### ISSUE THREAD

**Issue by mahmoudElfeel67** [facebook/facebook-android-sdk#1010]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [Github issues](https://github.com/facebook/facebook-android-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Java version

VERSION_1_8

### Android version

7.1.2 Nexus 6P

### Android SDK version

11.3.0 

### Installation platform & version

gradle

### Package

Login

### Goals

not to crash

### Expected results

not to crash

### Actual results

android.os.Parcel.readException (Parcel.java:1696)
android.os.Parcel.readException (Parcel.java:1645)
android.app.ActivityManagerProxy.startActivity (ActivityManagerProxy.java:3102)
android.app.Instrumentation.execStartActivity (Instrumentation.java:1520)
android.app.Activity.startActivityForResult (Activity.java:4229)
androidx.activity.ComponentActivity.startActivityForResult (ComponentActivity.java:597)
androidx.core.app.ActivityCompat.startActivityForResult (ActivityCompat.java:237)
androidx.activity.ComponentActivity$2.onLaunch (ComponentActivity.java:210)
androidx.activity.result.ActivityResultRegistry$3.launch (ActivityResultRegistry.java:226)
androidx.activity.result.ActivityResultLauncher.launch (ActivityResultLauncher.java:47)
androidx.fragment.app.FragmentManager.launchStartActivityForResult (FragmentManager.java:3000)
androidx.fragment.app.Fragment.startActivityForResult (Fragment.java:1424)
androidx.fragment.app.Fragment.startActivityForResult (Fragment.java:1398)
com.facebook.login.NativeAppLoginMethodHandler.tryIntent (NativeAppLoginMethodHandler.java:180)
com.facebook.login.KatanaProxyLoginMethodHandler.tryAuthorize (KatanaProxyLoginMethodHandler.java:71)
com.facebook.login.LoginClient.tryCurrentHandler (LoginClient.java:289)
com.facebook.login.LoginClient.tryNextHandler (LoginClient.java:242)
com.facebook.login.LoginClient.authorize (LoginClient.java:123)
com.facebook.login.LoginClient.startOrContinueAuth (LoginClient.java:104)
com.facebook.login.LoginFragment.onResume (LoginFragment.java:164)
androidx.fragment.app.Fragment.performResume (Fragment.java:3039)
androidx.fragment.app.FragmentStateManager.resume (FragmentStateManager.java:607)
androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:306)
androidx.fragment.app.FragmentStore.moveToExpectedState (FragmentStore.java:112)
androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1647)
androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:3128)
androidx.fragment.app.FragmentManager.dispatchResume (FragmentManager.java:3086)
androidx.fragment.app.FragmentController.dispatchResume (FragmentController.java:273)
androidx.fragment.app.FragmentActivity.onResumeFragments (FragmentActivity.java:458)
androidx.fragment.app.FragmentActivity.onPostResume (FragmentActivity.java:447)
android.app.Activity.performResume (Activity.java:6812)
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3432)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3495)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2758)
android.app.ActivityThread.-wrap12 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1496)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:154)
android.app.ActivityThread.main (ActivityThread.java:6187)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:889)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:779) 

### Steps to reproduce

login with facebook

### Code samples & details

```java
LoginManager.getInstance().logOut();
        LoginManager.getInstance().registerCallback(CallbackManager.Factory.create(), onFacebookLogin);
List<String> permissionNeeds = Arrays.asList("public_profile", "email");

if (fragment == null) {
    LoginManager.getInstance().logInWithReadPermissions(activity, permissionNeeds);
} else {
   LoginManager.getInstance().logInWithReadPermissions(fragment, permissionNeeds);
}
```

**Comment by linmx0130** [comment]
Hi @mahmoudElfeel67, thanks for raising this issue! Could you provide us with the exception content and how to perform the resuming?

**Comment by mahmoudElfeel67** [comment]
@linmx0130  the stack trace is in the issue itself , and i dont perform any resuming manually in the app this is all the facebook sdk work not our code , initiated when the user wants to sign in with facebook 
note it only happens on one device  7.1.2 Nexus 6P

**Comment by linmx0130** [comment]
@mahmoudElfeel67 Thanks for the information! I will have look into it.

**Comment by xDragonZ** [comment]
Using SDK 12.0.0 the crash started to appear, removing "onActivityResult" and "onResume" doesn't solve this issue.
To reproduce this we have to have a clean install of the app or didn't login with FB using the app recently.

```
java.lang.RuntimeException: Unable to resume activity {com.example.test/com.facebook.FacebookActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4899)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4936)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:224)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2317)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8595)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.List.size()' on a null object reference
        at com.facebook.login.KatanaProxyLoginMethodHandler.tryAuthorize(KatanaProxyLoginMethodHandler.java:71)
        at com.facebook.login.LoginClient.tryCurrentHandler(LoginClient.java:289)
        at com.facebook.login.LoginClient.tryNextHandler(LoginClient.java:242)
        at com.facebook.login.LoginClient.authorize(LoginClient.java:123)
        at com.facebook.login.LoginClient.startOrContinueAuth(LoginClient.java:104)
        at com.facebook.login.LoginFragment.onResume(LoginFragment.java:164)
        at androidx.fragment.app.Fragment.performResume(Fragment.java:3039)
        at androidx.fragment.app.FragmentStateManager.resume(FragmentStateManager.java:607)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:306)
        at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:112)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1647)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3128)
        at androidx.fragment.app.FragmentManager.dispatchResume(FragmentManager.java:3086)
        at androidx.fragment.app.FragmentController.dispatchResume(FragmentController.java:273)
        at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:458)
        at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:447)
        at android.app.Activity.performResume(Activity.java:8378)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4889)
```

**Comment by linmx0130** [comment]
Hi @CharlesTai, thanks for the report! We're still working on figuring out the root cause of this bug.

**Comment by tdrp** [comment]
Repros for me with slightly different stack trace. This is a new bug

implementation 'com.facebook.android:facebook-login:9.0.0'


Caused by java.lang.IllegalArgumentException
Unknown component: com.facebook.katana/com.facebook.katana.ProxyAuth

androidx.fragment.app.Fragment.startActivityForResult (Fragment.java:1266)
arrow_right
com.facebook.login.NativeAppLoginMethodHandler.tryIntent (NativeAppLoginMethodHandler.java:151)
com.facebook.login.KatanaProxyLoginMethodHandler.tryAuthorize (KatanaProxyLoginMethodHandler.java:67)
com.facebook.login.LoginClient.tryCurrentHandler (LoginClient.java:280)
com.facebook.login.LoginClient.tryNextHandler (LoginClient.java:233)
com.facebook.login.LoginClient.authorize (LoginClient.java:120)
com.facebook.login.LoginClient.startOrContinueAuth (LoginClient.java:101)
com.facebook.login.LoginFragment.onResume (LoginFragment.java:161)

**Comment by JenyChen** [comment]
I have face the same issue in the same device, but the facebook sdk version is 9.1.1
crash log :

Caused by java.lang.IllegalArgumentException: Unknown component: com.facebook.katana/com.facebook.katana.ProxyAuth
       at android.os.Parcel.readException(Parcel.java:1688)
       at android.os.Parcel.readException(Parcel.java:1637)
       at android.app.ActivityManagerProxy.startActivity(ActivityManagerProxy.java:3115)
       at android.app.Instrumentation.execStartActivity(Instrumentation.java:1520)
       at android.app.Activity.startActivityForResult(Activity.java:4229)
       at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:676)
       at androidx.core.app.ActivityCompat.startActivityForResult(ActivityCompat.java:234)
       at androidx.fragment.app.FragmentActivity.startActivityFromFragment(FragmentActivity.java:796)
       at androidx.fragment.app.FragmentActivity$HostCallbacks.onStartActivityFromFragment(FragmentActivity.java:933)
       at androidx.fragment.app.Fragment.startActivityForResult(Fragment.java:1206)
       at androidx.fragment.app.Fragment.startActivityForResult(Fragment.java:1194)
       at com.facebook.login.NativeAppLoginMethodHandler.tryIntent(NativeAppLoginMethodHandler.java:151)
       at com.facebook.login.KatanaProxyLoginMethodHandler.tryAuthorize(KatanaProxyLoginMethodHandler.java:67)
       at com.facebook.login.LoginClient.tryCurrentHandler(LoginClient.java:280)
       at com.facebook.login.LoginClient.tryNextHandler(LoginClient.java:233)
       at com.facebook.login.LoginClient.authorize(LoginClient.java:120)
       at com.facebook.login.LoginClient.startOrContinueAuth(LoginClient.java:101)
       at com.facebook.login.LoginFragment.onResume(LoginFragment.java:161)
       at androidx.fragment.app.Fragment.performResume(Fragment.java:2649)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:922)
       at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
       at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2659)
       at androidx.fragment.app.FragmentManagerImpl.dispatchResume(FragmentManagerImpl.java:2625)
       at androidx.fragment.app.FragmentController.dispatchResume(FragmentController.java:268)
       at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:479)
       at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:468)
       at android.app.Activity.performResume(Activity.java:6812)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3446)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3509)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2772)
       at android.app.ActivityThread.-wrap12(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1507)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6236)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:891)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:781)


have you fixed it in 12.0.0? @linmx0130

**Comment by GeorgeGribkov** [comment]
Hi everyone.

Has anybody managed to solve the issue? @linmx0130 have you found the root of the problem?

I'm having the same crash with the same call stack [as JenyJen ](https://github.com/facebook/facebook-android-sdk/issues/1010#issuecomment-1049463312).

**Device:** Moto G Turbo Edition
**Android:** 7.1.2

Thank you for your help!
